### DISCUSSION TITLE
Plain text secret in YAML

### DISCUSSION THREAD

**Discussion by Abhi5h3k** [StackStorm/st2#6201] [Q&A]
I'm new to StackStorm and noticed that secrets are shown as plain text in workflow actions. 

For example, I was using `core.remote_sudo` and saw my password visible in plain text in the YAML file, even though it is hidden in the UI. 

Is there a better way to manage passwords in StackStorm?

![image](https://github.com/StackStorm/st2/assets/37928721/0060dce2-5b71-49ad-8be3-8abcec4d8639)

**Comment by fdrab** [comment]
a better way is to use the internal datastore and then just <% st2kv("datastore_key_under_which_the_pw_is_stored") %> instead of hardcoding the pw into the yaml
EDIT: If you want to store the passwords in the datastore encrypted, don't forget to add ", decrypt=>true" to the st2kv

**Comment by Abhi5h3k** [comment]
Thank you @fdrab, I was able to resolve it.

**Leaving Doc links for future reference:**

1.[ Storing and Retrieving Key-Value Pairs via CLI](https://docs.stackstorm.com/datastore.html#referencing-key-value-pairs-in-action-definitions)
2. [Referencing Key-Value Pairs in Action Definitions](https://docs.stackstorm.com/datastore.html#referencing-key-value-pairs-in-action-definitions)
3. [Referencing Datastore Keys in Jinja]( https://docs.stackstorm.com/reference/jinja.html#referencing-datastore-keys-in-jinja)
5. [Securing Secrets (admin only)](https://docs.stackstorm.com/datastore.html#securing-secrets-admin-only)
